lib/HALSim/src/Device.h

```diff
@@ -115,6 +130,12 @@ class Device : public IDevice
      * Socket stream for communication with the device.
      */
     SocketClient m_socket;
+
+    /** Server Address. */
+    const char* m_address;
+
+    /** Server Port Number. */
+    const char* m_port;

```

Comment:
Why do we use a string instead of uint16_t?

Reply:
Because it is given as string from the command line, and winsock also needs it as string. Unnecessary conversion to and from uint